client/src/components/Search/index.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react';
+import { Select, Radio, Button } from 'antd';
+import world from 'full-countries-cities';
+
+import Keywords from '../Keywords';
+
+import './style.css';
+
+const { Option } = Select;
+
+class Search extends Component {
+  state = {
+    country: undefined,
+    city: undefined,
+    lookingFor: '',
+    keyword: '',
+    viewKeywords: false,
+  };
+
+  lookingFor = { STAY: 'stay', EAT: 'eat', SHOP: 'shop' };
+
+  renderOptions = list =>
+    list.map(item => (
+      <Option key={item} value={item}>
+        {item}
+      </Option>
+    ));
+
+  onSubmit = () => {};
+
+  setKeyword = value => {
+    this.setState({ keyword: value, viewKeywords: false });
+  };
+
+  toggleKeywordList = () => {
+    this.setState(state => ({ viewKeywords: !state.viewKeywords }));
+  };
+
+  dropdownFilter = (input, option) =>
+    option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0;
+
+  render() {
+    const { country, city, keyword, viewKeywords, lookingFor } = this.state;
+
+    const { EAT, SHOP, STAY } = this.lookingFor;
+    const cities = country ? world.getCities(country) : undefined;
```

Comment:
instead of `undefined` you can use an emty array `[]`. Then instead of
```js
{country && cities !== undefined && this.renderOptions(cities)}
```
you could just do `this.renderOptions(cities)`. I believe that should give the same result